feat: unifyquery 查询增加 proxy 代理层，兼容蓝鲸查询结构体 --story=126503120

Zhaoyikaiii opened the pull request.

shamcleren commented on pkg/unify-query/service/http/middleware/post_response.go:
```diff
@@ -0,0 +1,141 @@
+	}
+}
+
+func shouldReturnSuccess(data interface{}, hasData bool) bool {
```
不需要使用反射吧？直接参考之前的 error 才判断为异常，不然的话，就正常就行，即使数据为空，result 状态也是 true

Zhaoyikaiii replied:
已移除

shamcleren commented on pkg/unify-query/service/http/register_urls.go:
```diff
@@ -21,14 +22,32 @@ import (
 }
 
+func registerUnifiedHandler(handlerPath string) (func(c *gin.Context), bool) {
+	// format like: http.unifyProcess.query/ts/promql
```
这里要用这种格式吗？是否可以使用列表配置就行了？

Zhaoyikaiii replied:
已移除

shamcleren commented on pkg/unify-query/service/http/response.go:
```diff
@@ -32,6 +33,11 @@ func (r *response) failed(ctx context.Context, err error) {
 	user := metadata.GetUser(ctx)
 	metric.APIRequestInc(ctx, r.c.Request.URL.Path, metric.StatusFailed, user.SpaceUID, user.Source)
 
+	if _, ok := r.c.Get(middleware.ContextConfigUnifyResponseProcess); ok {
```
不要使用更改原接口的方式，要完全新写一个返回。

Zhaoyikaiii replied:
已修改

Zhaoyikaiii committed:
feat: unify response handling for API Gateway

- add unified response processing middleware and handler
- enhance success/failure response methods
- support multiple handlers per route and unify handler retrieval via metadata
- use json.RawMessage for request data; align handler mapping naming and config path
- initialize configuration loading and add POST request configuration
- improve error logging and ensure span closure

Zhaoyikaiii committed:
Merge branch 'master' into feat/post_response_unify_handler/#1010158081126503120

Zhaoyikaiii committed:
feat: unify response handling middleware for consistent API responses

Zhaoyikaiii committed:
chore: remove trace id

Zhaoyikaiii committed:
chore: remove pre-resp process func

Zhaoyikaiii committed:
chore: remove dafaFieldName

shamcleren commented on pkg/unify-query/service/http/hook.go:
```diff
@@ -62,6 +62,7 @@ func setDefaultConfig() {
 
 	viper.SetDefault(CheckQueryTsConfigPath, "/check/query/ts")
 	viper.SetDefault(CheckQueryPromQLConfigPath, "/check/query/ts/promql")
+	viper.SetDefault(ApiGwConfigPath, "/query/apigw")
```
这里改成  /proxy

Zhaoyikaiii replied:
已修改

shamcleren commented on pkg/unify-query/service/http/register_urls.go:
```diff
@@ -42,6 +43,8 @@ func (r *RegisterHandlers) register(method, handlerPath string, handlerFunc ...g
 
 	log.Infof(r.ctx, "registerHandlers => [%s] %s", method, handlerPath)
+
+	metadata.AddHandler(handlerPath, handlerFunc...)
```
拆分一下 register，写一个 proxyRegister

Zhaoyikaiii replied:
已修改

Zhaoyikaiii committed:
feat: rename api_gw.go to proxy.go and unify response handling in registration

Zhaoyikaiii committed:
chore: change proxy handlerPath

Zhaoyikaiii committed:
feat: unify response handling by introducing RegisterHandler for API registration

shamcleren commented on pkg/unify-query/service/http/api/response.go:
```diff
@@ -35,6 +36,12 @@ func (r *response) failed(ctx context.Context, err error) {
 	user := metadata.GetUser(ctx)
 	metric.APIRequestInc(ctx, r.c.Request.URL.Path, metric.StatusFailed, user.SpaceUID, user.Source)
+
+	if _, ok := r.c.Get(proxy.ContextConfigUnifyResponseProcess); ok {
```
这里增加一下注释，说明什么情况下会进入到这段逻辑

shamcleren commented on pkg/unify-query/service/http/api/response.go:
```diff
@@ -44,5 +51,15 @@ func (r *response) success(ctx context.Context, data interface{}) {
 	log.Debugf(ctx, "query data size is %s", fmt.Sprint(unsafe.Sizeof(data)))
 	user := metadata.GetUser(ctx)
 	metric.APIRequestInc(ctx, r.c.Request.URL.Path, metric.StatusSuccess, user.SpaceUID, user.Source)
+	isUnifyRespProcess := r.isConfigUnifyRespProcess(r.c)
```
同上

shamcleren commented on pkg/unify-query/service/http/api/response.go:
```diff
@@ -44,5 +51,15 @@ func (r *response) success(ctx context.Context, data interface{}) {
 	user := metadata.GetUser(ctx)
 	metric.APIRequestInc(ctx, r.c.Request.URL.Path, metric.StatusSuccess, user.SpaceUID, user.Source)
+	isUnifyRespProcess := r.isConfigUnifyRespProcess(r.c)
+	if isUnifyRespProcess {
```
这里可以直接简化成 if r.isConfigUnifyRespProcess(r.c) 了

shamcleren commented on pkg/unify-query/service/http/endpoint/register.go:
```diff
@@ -0,0 +1,68 @@
+}
+
+func (r *RegisterHandler) Register(method, handlerPath string, handlerFunc ...gin.HandlerFunc) {
+	metadata.AddHandler(handlerPath, handlerFunc...)
```
这里注释也要写清楚一点

shamcleren commented on pkg/unify-query/service/http/endpoint/register.go:
```diff
@@ -0,0 +1,68 @@
+	}
+}
+
+func (r *RegisterHandler) RegisterWithoutRecord(method, handlerPath string, handlerFunc ...gin.HandlerFunc) {
```
这个函数是不是明显可以合并到上面 Register 里面？

Zhaoyikaiii committed:
feat: refactor RegisterHandler methods for unified response handling

shamcleren committed:
feat: post_response_unify_handler #1010158081126503120

shamcleren committed:
feat: post_response_unify_handler #1010158081126503120
